src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStoreConfig.java

```diff
@@ -0,0 +1,163 @@
+ * the change of container does not affected to the Artifact functionality, you could change the container
+ * and it would still work if both container contains the same data.
+ */
+@Extension
+public class S3BlobStoreConfig extends GlobalConfiguration {
+
+    private static final String BUCKET_REGEXP = "^([a-z]|(\\d(?!\\d{0,2}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})))([a-z\\d]|(\\.(?!(\\.|-)))|(-(?!\\.))){1,61}[a-z\\d\\.]$";
+    private static final Pattern bucketPattern = Pattern.compile(BUCKET_REGEXP);
+
+    private static final Logger LOGGER = Logger.getLogger(S3BlobStoreConfig.class.getName());
+
+    @SuppressWarnings("FieldMayBeFinal")
+    private static boolean DELETE_ARTIFACTS = Boolean.getBoolean(S3BlobStoreConfig.class.getName() + ".deleteArtifacts");
+    @SuppressWarnings("FieldMayBeFinal")
+    private static boolean DELETE_STASHES = Boolean.getBoolean(S3BlobStoreConfig.class.getName() + ".deleteStashes");
+
+    /**
+     * Name of the S3 Bucket.
+     */
+    private String container;
+    /**
+     * Prefix to use for files, use to be a folder.
+     */
+    private String prefix;
+    /**
+     * force the region to use for the URLs generated.
+     */
+    private String region;
+
+    @DataBoundConstructor
+    public S3BlobStoreConfig() {
+        load();
+    }
+
+    public String getContainer() {
+        return container;
+    }
+
+    @DataBoundSetter
+    public void setContainer(String container) {
+        this.container = container;
+        save();
+    }
+
+    public String getPrefix() {
+        return prefix;
+    }
+
+    @DataBoundSetter
+    public void setPrefix(String prefix) {
+        this.prefix = prefix;
+        save();
+    }
+
+    public String getRegion() {
+        return region;
+    }
+    @DataBoundSetter
+    public void setRegion(String region) {
+        this.region = region;
+        save();
+    }
+
+    public boolean isDeleteArtifacts() {
+        return DELETE_ARTIFACTS;
+    }
+
+    public boolean isDeleteStashes() {
+        return DELETE_STASHES;
+    }
+
+    @Nonnull
+    @Override
+    public String getDisplayName() {
+        return "Amazon S3 Bucket Access settings";
+    }
+
+    @Nonnull
+    public static S3BlobStoreConfig get() {
+        return ExtensionList.lookupSingleton(S3BlobStoreConfig.class);
+    }
+
+    public ListBoxModel doFillRegionItems() {
+        ListBoxModel regions = new ListBoxModel();
+        regions.add("Auto", "");
+        for (String s : Region.DEFAULT_S3) {
+            regions.add(s);
+        }
+        return regions;
+    }
+
+    public FormValidation doCheckContainer(@QueryParameter String container){
+        FormValidation ret = FormValidation.ok();
+        if (StringUtils.isBlank(container)){
+            ret = FormValidation.warning("The container name cannot be empty");
+        } else if (!bucketPattern.matcher(container).matches()){
+            ret = FormValidation.error("The S3 Bucket name does not match with S3 bucket rules");
```

Comment:
🐛 `does not match S3 bucket rules`

@kuisathaverat

Reply:
noted, I will commit it in the next PR